qemu-check.exp

```diff
@@ -75,11 +75,11 @@ proc check_test_result arg {
 		# Exit when result separator is seen
 		"+-----------------------------------------------------\r\r" {}
 		# Handle errors in TEE core output
-		-i $arg -re {(^..TC:.*assertion.*failed at.*)\n} {
+		-i $arg -re {(^..TC:[^\n]*assertion.*failed at.*)\n} {
```

Comment:
Why do we have a `\n` at the end, wouldn't it make more sense with `$`?

Reply:
Very good question which made me realize that in Expect, `^` and `$` are not the beginning and end of a line, but of the matching buffer. Therefore, the `\n` at the end of the RE is indeed correct, but the `^` at the start is not.
So the proper RE is `\n(..TC:[^\n]*assertion[^\n]*failed at[^\n]*)\n`.
I'll fix.